cfme/tests/control/test_alerts.py

```diff
@@ -121,25 +192,182 @@ def test_alert_vm_turned_on_more_than_twice_in_past_15_minutes(
+
+    Metadata:
+        test_flag: alerts, provision, metrics_collection
+    """
+    email = fauxfactory.gen_email()
+    alert = explorer.Alert(
+        "Trigger by CPU {}".format(fauxfactory.gen_alpha(length=4)),
+        active=True,
+        based_on="VM and Instance",
+        evaluate=(
+            "Real Time Performance",
+            {
+                "performance_field": "CPU - % Used",
+                "performance_field_operator": ">",
+                "performance_field_value": "0",
+                "performance_trend": "Don't Care",
+                "performance_time_threshold": "3 Minutes",
+            }),
+        notification_frequency="5 Minutes",
+        emails=email,
+    )
+    alert.create()
+    request.addfinalizer(alert.delete)
+
+    setup_for_alerts(request, [alert])
+    wait_for_alert(smtp_test, alert, delay=30 * 60, additional_checks={
+        "text": vm_name, "from_address": email})
+
+
+@pytest.mark.uncollectif(lambda provider: provider.type not in CANDU_PROVIDER_TYPES)
+def test_alert_timeline_cpu(request, vm_name, provider, ssh, vm_crud):
+    """ Tests a custom alert that uses C&U data to trigger an alert. It will run a script that makes
+    a CPU spike in the machine to trigger the threshold. The alert is displayed in the timelines.
+
+    Metadata:
+        test_flag: alerts, provision, metrics_collection
+    """
+    alert = explorer.Alert(
+        "TL event by CPU {}".format(fauxfactory.gen_alpha(length=4)),
+        active=True,
+        based_on="VM and Instance",
+        evaluate=(
+            "Real Time Performance",
+            {
+                "performance_field": "CPU - % Used",
+                "performance_field_operator": ">",
+                "performance_field_value": "20",
+                "performance_trend": "Don't Care",
+                "performance_time_threshold": "2 Minutes",
+            }),
+        notification_frequency="5 Minutes",
+        timeline_event=True,
+    )
+    alert.create()
+    request.addfinalizer(alert.delete)
+
+    setup_for_alerts(request, [alert])
+    # Generate a 100% CPU spike for 5 minutes, that should be noticed by CFME.
+    ssh.cpu_spike(seconds=5 * 60, cpus=4)
+
+    def _timeline_event_present():
+        vm_crud.open_timelines()
+        select = pytest.sel.Select("//select[@name='tl_fl_grp2']")  # TODO: Make a timelines module?
+        pytest.sel.select(select, "Alarm/Status change/Errors")
+        for event in jstimelines.events():
+            info = event.block_info()
+            if info.get("Event Type") != "EVMAlertEvent":
+                continue
+            if info.get("Event Source") != "MiqAlert":
+                continue
+            if info["Source VM"] == vm_name:
+                return True
+        return False
+
+    wait_for(_timeline_event_present, num_sec=30 * 60, delay=15, message="timeline event present")
+
+
+@pytest.mark.skipif(True, reason="SNMP hogs the tests.")
```

Comment:
@psav  SNMP installation hogged the tests so I left it for later to be resolved.